deathnode/constraints_test.go

```diff
@@ -63,15 +63,48 @@ func TestProtectedConstraint(t *testing.T) {
 	})
 }
 
-func prepareMonitorsForConstraints(awsConn *aws.ConnectionMock, mesosConn *mesos.ClientMock) (*monitor.AutoscalingGroupMonitor, *monitor.MesosMonitor) {
+func TestFilterFrameworkContraint(t *testing.T) {
+
+	Convey("When creating a filterFrameworkContraint", t, func() {
+		awsConn := &aws.ConnectionMock{
+			Records: map[string]*[]string{
+				"DescribeInstanceById": {"node1", "node2", "node3"},
+				"DescribeAGByName":     {"default"},
+			},
+		}
+		mesosConn := &mesos.ClientMock{
+			Records: map[string]*[]string{
+				"GetMesosFrameworks": {"default"},
+				"GetMesosSlaves":     {"default"},
+				"GetMesosTasks":      {"default"},
+			},
+		}
+		instanceMonitor, mesosMonitor := prepareMonitorsForConstraints(awsConn, mesosConn, []string{"frameworkName1", "frameworkName2"})
+		mesosMonitor.Refresh()
+
+		Convey("it should filter instances with tasks running those frameworks", func() {
+			constraint, _ := newConstraint("filterFrameworkConstraint=frameworkName1,frameworkName2")
+			instances := constraint.filter(instanceMonitor.GetInstances(), mesosMonitor)
+			So(len(instances), ShouldEqual, 3)
+
+			constraint, _ = newConstraint("filterFrameworkConstraint=frameworkName1")
+			instances = constraint.filter(instanceMonitor.GetInstances(), mesosMonitor)
+			So(len(instances), ShouldEqual, 1)
+		})
+	})
+}
+
+func prepareMonitorsForConstraints(
+	awsConn *aws.ConnectionMock, mesosConn *mesos.ClientMock, protectedFrameworks []string) (
+	*monitor.AutoscalingGroupMonitor, *monitor.MesosMonitor) {
 
 	ctx := &context.ApplicationContext{
 		AwsConn:   awsConn,
 		MesosConn: mesosConn,
 		Conf: context.ApplicationConf{
 			DeathNodeMark:            "DEATH_NODE_MARK",
 			AutoscalingGroupPrefixes: []string{"some-Autoscaling-Group"},
-			ProtectedFrameworks:      []string{"frameworkName1"},
+			ProtectedFrameworks:      protectedFrameworks,
```

Comment:
nit: rename to something like protectedFrameworksList for readability

nit == not blocker to merge just style

Reply:
👍